modules/statistics/RTPStatsCollector.js

```diff
@@ -638,13 +643,36 @@ StatsCollector.prototype.processStatsReport = function() {
 
             if (codec) {
                 /**
-                 * The mime type has the following form: video/VP8 or audio/ISAC,
-                 * so we what to keep just the type after the '/', audio and video
-                 * keys will be added on the processing side.
+                 * The mime type has the following form: video/VP8 or audio/ISAC, so we what to keep just the type
+                 * after the '/', audio and video keys will be added on the processing side.
                  */
                 const codecShortType = codec.mimeType.split('/')[1];
 
                 codecShortType && ssrcStats.setCodec(codecShortType);
+
+                // Calculate the encodeTime stat for outbound video streams.
+                const track = this.peerconnection.getTrackBySSRC(ssrc, false);
+
+                if (now.type === 'outbound-rtp'
+                    && now.active
+                    && track?.isVideoTrack()
+                    && this.peerconnection.usesCodecSelectionAPI()
+                    && before?.totalEncodeTime
+                    && before?.framesEncoded) {
+                    const encodeTimeDelta = now.totalEncodeTime - before.totalEncodeTime;
+                    const framesEncodedDelta = now.framesEncoded - before.framesEncoded;
+                    const encodeTimePerFrameInMs = 1000 * encodeTimeDelta / framesEncodedDelta;
+                    const encodeTimeStats = {
+                        codec: codecShortType,
+                        encodeTime: encodeTimePerFrameInMs,
+                        isBandwidthLimited: now.qualityLimitationReason === 'bandwidth',
```

Comment:
Why not surface the reason raw? Why do we not care anout the `cpu` reason, for example?

Reply:
We could and I actually modified this to surface `qualityLimitationReason` in the patch that I am working on now. Will update it here :)